docs/reference/local_development.md

```diff
@@ -0,0 +1,198 @@
+# Local Development
+
+*This document is a conceptual overview of how Tye behaves when using `tye run` for local development. For reference docs on the `tye run` command see [here](/docs/reference/commandline/tye-run.md).*
+
+Tye acts as a *local orchestrator* for development by building and launching multiple services, managing and monitoring services, and enabling network communication between services.
+
+This document will describe how these features work, as well as the implications and behaviors of various optional settings.
+
+## Executing `tye run`
+
+When executing `tye run` for local development, Tye goes through the following steps:
+
+- Load the application (`tye.yaml`)
+- Build all projects 
+- Pull all container images
+- Assign ports and bindings
+- Launch and monitor all services
+
+## Loading the application
+
+A Tye application is specified using `tye.yaml`. You can find more information about the Tye application format [here](/docs/reference/schema.md).
```

Comment:
Let's figure out a way to share this info with tye deploy. I would end up duplicating most of this.